packages/blade/src/components/BottomSheet/BottomSheet.tsx

```diff
@@ -0,0 +1,413 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable react/jsx-no-useless-fragment */
+/* eslint-disable consistent-return */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import React from 'react';
+import styled from 'styled-components';
+import { rubberbandIfOutOfBounds, useDrag } from '@use-gesture/react';
+import usePresence from 'use-presence';
+import { BottomSheetGrabHandle, BottomSheetHeader } from './BottomSheetHeader';
+import { BottomSheetBody } from './BottomSheetBody';
+import type { SnapPoints } from './utils';
+import { computeMaxContent, computeSnapPointBounds } from './utils';
+import { BottomSheetBackdrop } from './BottomSheetBackdrop';
+import { BottomSheetContext, useDropdownBottomSheetContext } from './BottomSheetContext';
+import { ComponentIds } from './componentIds';
+import { BottomSheetCloseButton } from './BottomSheetCloseButton';
+import BaseBox from '~components/Box/BaseBox';
+import { makeMotionTime, makeSpace } from '~utils';
+
+import { useScrollLock } from '~src/hooks/useScrollLock';
+import { useWindowSize } from '~src/hooks/useWindowSize';
+import { useIsomorphicLayoutEffect } from '~src/hooks/useIsomorphicLayoutEffect';
+import { assignWithoutSideEffects } from '~src/utils/assignWithoutSideEffects';
+import { useTheme } from '~components/BladeProvider';
+
+type BottomSheetProps = {
+  isOpen?: boolean;
+  onDismiss?: () => void;
+  children: React.ReactNode;
+  initialFocusRef?: React.MutableRefObject<any>;
+  snapPoints: SnapPoints;
+};
+
+export const BOTTOM_SHEET_EASING = 'cubic-bezier(.15,0,.24,.97)';
+
+const BottomSheetSurface = styled.div<{
+  windowHeight: number;
+  isDragging: boolean;
+}>(({ theme, windowHeight, isDragging }) => {
+  const offsetX = theme.shadows.offsetX.level[1];
+  const offsetY = theme.shadows.offsetY.level[1];
+  const blur = theme.shadows.blurRadius.level[1];
+  const shadowColor = theme.shadows.color.level[1];
+
+  const shadow1 = `${offsetX}px ${offsetY}px ${blur}px 0px ${shadowColor}`;
+  const shadow2 = `0px 0px 1px 0px ${shadowColor}`;
```

Comment:
Can't really think of better names, maybe shadowLayer1, shadowLayer2, but these shadows are built like this to match the figma styles which uses multiple styles since our tokens doesn't represent composite shadow tokens. 
Unless we create makeShadowToken util.